### ISSUE TITLE
needless_lifetimes: false positive with async fn

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/rust-clippy#3988]
Minimal repro:

```rust
#![feature(futures_api)]

pub struct Foo;
impl Foo {
    pub async fn foo(&mut self) {
    }
}
```

([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f772f09cb29b32a388af556a4adaf1fd)), gives (with clippy `0.0.212 (2019-04-14 fbb3a47)`)

```
warning: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
 --> src/lib.rs:5:5
  |
5 | /     pub async fn foo(&mut self) {
6 | |     }
  | |_____^
  |
  = note: #[warn(clippy::needless_lifetimes)] on by default
```

**Comment by oli-obk** [comment]
We need to ignore late bound lifetimes in the `needless_lifetimes` lint.

**Comment by realcr** [comment]
I think that a similar problems happens on my side:

```rust
error: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
  --> components/common/src/state_service.rs:37:5
   |
37 | /     pub async fn request_state(&mut self) -> Result<StateResponse<ST, MU>, StateClientError> {
38 | |         let (response_sender, response_receiver) = oneshot::channel();
39 | |         let state_request = StateRequest { response_sender };
40 | |
...  |
44 | |         Ok(await!(response_receiver).map_err(|_| StateClientError::ReceiveResponseError)?)
45 | |     }
   | |_____^
```
rustc version:
```bash
$ rustc --version
rustc 1.36.0-nightly (af98304b9 2019-05-11)
```
